### ISSUE TITLE
http://j.gs/3DYv

### ISSUE THREAD

**Issue by azedlal12** [reek/anti-adblock-killer#2891]
uBloc Origin detected on http://j.gs/3DYv

![ubloc](https://cloud.githubusercontent.com/assets/24457065/23339298/bb272a08-fc1f-11e6-8d41-291ca4802d4c.JPG)

**Comment by uBlock-user** [comment]
force update all filters, already patched

**Comment by uBlock-user** [comment]
@X01X012013 would you look into this ? The AAB is evaled but blocking eval also breaks the timer.

**Comment by jspenguin2017** [comment]
@uBlock-user I have the solution for `ay.gy`, it also skips countdown. I'll have a look at this one. 
```JavaScript
if (a.domCmp(["ay.gy"])) {
    //Disable open() before page starts to load and set abgo to an empty function when the page loads
    a.readOnly("open", function () { });
    a.on("load", function () {
        a.win.abgo = function () { };
    });
    //Skip countdown
    const _setInterval = a.win.setInterval;
    a.win.setInterval = function (func) {
        return _setInterval(func, 10);
    };
}
```

**Comment by uBlock-user** [comment]
> it also skips countdown

That would be nice, if you could do the same for this domain too.

**Comment by jspenguin2017** [comment]
@uBlock-user OK, this is the `eval`ed code: http://pastebin.com/FEHEdHqH
Well... Looks like we'll need a custom tool to decompile this...

**Comment by uBlock-user** [comment]
btw for popups, lock `window.admvpu` variable.

**Comment by uBlock-user** [comment]
this works :  
````
j.gs###fade
j.gs###_bd

if (a.domCmp(["ay.gy", "j.gs"])) {
    a.readOnly("open", function () { });
    a.on("load", function () {
        a.win.abdo = function () { };
    });
    const _setInterval = a.win.setInterval;
    a.win.setInterval = function (func) {
        return _setInterval(func, 10);
    };
}

**Comment by jspenguin2017** [comment]
The countdown gets stuck at 1s...

**Comment by uBlock-user** [comment]
Well, it didn't on my side and I got Skip AD button instantly.

**Comment by jspenguin2017** [comment]
Hum, do you have other custom filters?

**Comment by uBlock-user** [comment]
Nevermind, now it gets stuck at 1 sec for me too. Dont know how I was able to get the SKIP AD button previously.

**Comment by jspenguin2017** [comment]
Man that's some next level obfuscation. There are "URL expander" online, how do they work...

**Comment by uBlock-user** [comment]
haha, can tell those guys are desperate for some traffic.

I have never heard of expander, rather I have seen shortener which is quite the contrary.

**Comment by jspenguin2017** [comment]
@uBlock-user 
URL expanders: 
http://urlex.org/
http://deadfly.crazytec.net/ <- Doesn't seem to give the correct URL...

**Comment by jspenguin2017** [comment]
These guys are using bit wise operations to hide strings, there must be another way...

**Comment by uBlock-user** [comment]
Yes those ones, I remember. Actually, I have adf.ly and their whole shit network blacklisted, so I never use them, making people wait for 5 seconds is something I hate and I prefer direct links rather than going through a clusterfuck of third-party services wasting your time for some ad revenue. Sorry man, can't help you any further on this one.

**Comment by jspenguin2017** [comment]
@uBlock-user 
Well, at least we can skip the 5 sec countdown... Until they have implementation on server side to force the client to respect the countdown... like Cloudflare... 
At the end of the day, users only have control over their own devices. We open source volunteers, after all, ~~can't~~ can fight paid professionals whose only job is to shove ads down your throat. 
Their obfuscation is the best I've seen so far, I'm wondering what tool they used to do that...

**Comment by uBlock-user** [comment]
Your script 

+

````
j.gs###_bd
j.gs###fade
||static.adf.ly^$script,important

That should patch it up for good.

**Comment by jspenguin2017** [comment]
They seems to have anti-view-source mechanism. I know sometimes you see: 
```JavaScript
			function counter() {
				countdown -= 1;

			    $('#countdown').html(countdown);

				if (countdown <= 0) {
					$('#continue').show();
					$('#wait').hide();
					clearInterval(t);
				}
			}
```
But that is actually fake. 

![image](https://cloud.githubusercontent.com/assets/7283682/23344565/09372682-fc3c-11e6-91a6-e616ce2b2e0e.png)

**Comment by jspenguin2017** [comment]
Also it seems their website is polymorphic, if you refresh view-source a couple times, you'll see the source changes.

**Comment by uBlock-user** [comment]
> their website is polymorphic, if you refresh view-source a couple times, you'll see the source changes.

This answer why https://github.com/reek/anti-adblock-killer/issues/2891#issuecomment-282586782 worked. Good find.

**Comment by jspenguin2017** [comment]
Well, ~~I... surrender...~~ I will NEVER surrender! I recall that `eval` is called multiple times when the page launches, I bet 2 or 3 of them are decoys. 
Blocking the whole `adf.ly` network seems to be a wise choice, considering most `adf.ly` links are either BS or scam. Look on Yahoo Answers, so many bots abuse this. 

BTW your solution https://github.com/reek/anti-adblock-killer/issues/2891#issuecomment-282594498 doesn't work for me.

**Comment by jspenguin2017** [comment]
An update: Looks like my script still partially works. I think `j.gs` and `q.gs` are "premium" hard blocking mode...

Test links: 

http://adf.ly/1XHJoS
http://adf.ly/249cf
http://adf.ly/249hE

http://ay.gy/1kXaTE

http://j.gs/3DYv
http://j.gs/8evu

http://q.gs/AtRuw

**Comment by uBlock-user** [comment]
Yeah your partial solution should be good enough, not worth wasting any more time on this one.

**Comment by jspenguin2017** [comment]
@uBlock-user We got a (fully functional... at least for now) solution, it is added to 6.64. Many thanks to @ghajini and developers of AdsBypasser! Looks like the power of open source volunteers are way higher than I thought. 
```JavaScript
if (a.domCmp(["adf.ly", "ay.gy", "j.gs", "q.gs"])) {
    //Based on: AdsBypasser
    //License: https://github.com/adsbypasser/adsbypasser/blob/master/LICENSE
    a.doc.write = function () { };
    a.win.btoa = function () { };
    a.on("DOMContentLoaded", function () {
        a.win.cookieCheck = function () { };
        let encodedURL = a.doc.head.innerHTML.match(/var eu = '(?!false)(.*)'/)[1];
        const index = encodedURL.indexOf('!HiTommy');
        if (index >= 0) {
            encodedURL = encodedURL.substring(0, index);
        }
        let var1 = "", var2 = "";
        for (let i = 0; i < encodedURL.length; ++i) {
            if (i % 2 === 0) {
                var1 = var1 + encodedURL.charAt(i);
            } else {
                var2 = encodedURL.charAt(i) + var2;
            }
        }
        let decodedURL = a.win.atob(var1 + var2);
        decodedURL = decodedURL.substr(2);
        if (a.win.location.hash) {
            decodedURL += a.win.location.hash;
        }
        a.win.onbeforeunload = null;
        a.win.location.href = decodedURL;
    });
}
```

**Comment by uBlock-user** [comment]
Good one, works better than expected. This one directly fetches the link and then bypasses the adf.ly shit entirely within seconds.
